remove/remove.go

```diff
@@ -0,0 +1,199 @@
+package remove
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/wtsi-hgi/go-softpack-builder/build"
+	"github.com/wtsi-hgi/go-softpack-builder/config"
+	"golang.org/x/sys/unix"
+)
+
+type Error string
+
+func (e Error) Error() string {
+	return string(e)
+}
+
+type coreResponse struct {
+	Data struct {
+		DeleteEnvironment struct {
+			Message string `json:"message,omitempty"`
+			Path    string `json:"path,omitempty"`
+			Name    string `json:"name,omitempty"`
+		} `json:"deleteEnvironment"`
+	} `json:"data"`
+}
+
+type S3Remover interface {
+	RemoveFile(string) error
+}
+
+const graphQLDeleteEnvironment = `mutation ($name: String!, $envPath: String!) {
+	deleteEnvironment(
+			name: $name
+			path: $envPath
+	) {
+		... on DeleteEnvironmentSuccess {
+				message
+		}
+		... on EnvironmentNotFoundError {
+				message
+				path
+				name
+		}
+	}
+}`
+
+const graphQLEndpoint = "/graphql"
+
+type graphQLDeleteEnvironmentMutation struct {
+	Query     string `json:"query"` // graphQLDeleteEnvironment
+	Variables struct {
+		Name    string `json:"name"`
+		EnvPath string `json:"envPath"`
+	} `json:"variables"`
+}
+
+func Remove(conf *config.Config, s S3Remover, envPath, version string) error {
```

Comment:
Docs for public things please. Don't like "s" as a var name since single letters are usually used to reference the struct a method is for. s3r?